PhotoMap/Features/MapPhoto/ViewModels/MapPhotoViewModel.swift

```diff
@@ -68,14 +82,47 @@ class MapPhotoViewModel: NSObject, MapPhotoViewModelType {
             .assign(to: \.isHiddenCategoryPicker, on: self)
             .store(in: cancelBag)
     }
+
+    // MARK: - Utils
+    private func completionHandler(with completion: Subscribers.Completion<FirestoreError>) {
+        switch completion {
+        case .failure(let error):
+            coordinator.errorAlertSubject.send(error)
+        case .finished:
+            return
+        }
+    }
+
+    private func saveNewPhoto(with description: String) {
+        guard let data = photoPublisher.image.pngData() else {
+            coordinator.errorAlertSubject.send(.custom("Image error"))
+            return
+        }
+        photoPublisher.description = description
+
+        firestoreService.uploadPhoto(data)
+            .receive(on: DispatchQueue.main)
+            .trackActivity(activityIndicator)
+            .flatMap { [unowned self] url in
+                self.firestoreService.addUserMarker(with: self.photoPublisher.toDictionary(urls: [url.absoluteString]))
+            }
```

Comment:
I suggest to combine this two methods into one in the service so the view model only call one method without flatMap and receive result. This will encapsulate logic as well